cleanlab/object_detection/filter.py

```diff
@@ -0,0 +1,128 @@
+# Copyright (C) 2017-2023  Cleanlab Inc.
+# This file is part of cleanlab.
+#
+# cleanlab is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published
+# by the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# cleanlab is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with cleanlab.  If not, see <https://www.gnu.org/licenses/>.
+
+"""Methods to find label issues in an object detection dataset (object detection data), where each annotated bounding box in an image receives its own class label."""
+
+from typing import List, Any, Dict
+import numpy as np
+
+from cleanlab.object_detection.rank import (
+    _get_valid_inputs_for_compute_scores,
+    _compute_overlooked_box_scores,
+    _compute_badloc_box_scores,
+    _compute_swap_box_scores,
+)
+
+
+def find_label_issues(
+    labels: List[Dict[str, Any]],
+    predictions: List[np.ndarray],
+) -> np.ndarray:
+    """
+    Identifies potentially mislabeled examples in an object detection dataset.
+    An example is flagged as with a label issue if *any* of the boxes appear to be incorrectly annotated for this example.
+
+    Parameters
+    ----------
+    labels:
+        A list of ``N`` dictionaries such that ``labels[i]`` contains the given labels for the `i`-th image in the format
+       ``{'bboxes': np.ndarray((M,4)), 'labels': np.ndarray((M,)), 'image_name': str}`` where ``L`` is the number of annotated bounding boxes
+       for the `i`-th image and ``bboxes[j]`` is in the format ``[x,y,x,y]`` with given label ``labels[j]``. (``image_name`` is optional here)
```

Comment:
Are these in COCO format? I'd explicitly state the name of the famous dataset that uses the same format, so somebody can more easily look up tutorials about it.

Reply:
@aditya1503 can you suggest where to best learn about this format? Seems to be mmdet and detectron2 train format but I am less familiar with the training.

Reply:
[x,y,w,h] is what's followed, that's standard across many datasets, but while predicting, most models predict [x,y,x,y]

Reply:
suggestion:
```
 Each row in the matrix represents a predicted bounding box and contains five columns representing [x1,y1,x2,y2,pred_prob]. Here, [x1,y1] corresponds to the coordinates of the bottom-left corner of the bounding box, while [x2,y2] corresponds to the coordinates of the top-right corner of the bounding box. The last column, pred_prob, represents the predicted probability that the bounding box contains an object of the class k.
```

Reply:
Here's a good link on formatting data: https://mmdetection.readthedocs.io/en/dev-3.x/advanced_guides/customize_dataset.html
(COCO style)